### PR TITLE
Fix #2195 add interface getConfigInfo getConfigToProperties and add content parse properties ContentUtils.toProperties() Support (xml,json,yaml,properties)

### LINKED ISSUE #2195: Client should support get Data Configuration format
when we add data to chose Data Configuration format,but client get the data we not Judge the data format （text、JSON、XML and so on）.

### PR DESCRIPTION
**Pull request by lidaoyang**
1、add interface getConfigInfo return ConfigInfo(dataId, group, tenant, type, appName,content)，When get configinfo  from the Snapshot, the type is text. 
2、add interface  getConfigToProperties return properties object ,Convert content to properties ,Support (xml,json,yaml,properties)
3、add content parse properties ContentUtils.toProperties() Support (xml,json,yaml,properties)

## What is the purpose of the change

XXXXX

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [x] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: client/src/main/java/com/alibaba/nacos/client/config/utils/ConfigParseUtils.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+/**
+ * ConfigParseUtils.
+ *
+ * @author Nacos
+ */
+public final class ConfigParseUtils {
+    
+    private static final String LINK_CHAR = "#@#";
+    
+    @SuppressWarnings("checkstyle:StaticVariableName")
```
[the remark this answers is not in the document]
Comment (lidaoyang) [reply client/src/main/java/com/alibaba/nacos/client/config/utils/ConfigParseUtils.java]: OK

**Comment by lidaoyang** [comment]
> 2010？
> Is it copied from others?

copied from nacos-spring-context

**Comment by lidaoyang** [comment]
> Sorry, We can't merge this PR.
> Because it seems that copy from others.
> 
> And some classes seems that has been removed or refactored, and you want to add them back.
> 
> I think we should discuss more for this Issue and PR.

Yes, when I saw the nacos-spring-context source code, I found that I converted the content into a tool class of properties, which is very practical in nacos-client and is similar to the requirement of issue #2195, so I migrated it.

是的，我在看到nacos-spring-context 源码时发现把content转换为properties的工具类，这个在nacos-client中实用性很高，而且和issue #2195的需求类似，所以我把它迁移过来了。

**Comment by chuntaojun** [comment]
first you can provide the conversion tool for external usesecond don't change ConfigService for now

**Comment by lidaoyang** [comment]
> my option：
> first you can provide the conversion tool for external use
> second don't change ConfigService for now

I see what you mean. Put this in the client toolkit instead of using it for special configuration content conversion, right? 
Why can't ConfigService be changed now?

我明白你的意思，把这个放在client工具包中，不要用在专门的配置内容转换上面是吗？
现在ConfigService为什么不能更改呢？

**Comment by lidaoyang** [comment]
> my option：
> first you can provide the conversion tool for external use
> second don't change ConfigService for now

I submit another commit that only keeps the tool classes that convert the configuration content to properties. Which classes in the picture?
我再提交一个commit 只保留把配置内容转换为properties的工具类，图片中的那些类？
image

**Comment by KomachiSion** [comment]
@lidaoyang I think we need has more discussion about the issue and PR.

@TsingLiang @chuntaojun Please task some time to discuss with @lidaoyang about this feature.

**Pull request closed without merging**